frontend/src/components/table.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { useState } from 'react';
+import '../styles/table.css';
+
+interface driversSettings {
+  name: string;
+  netid: string;
+  email: string;
+  phone: string;
+}
+
+interface FormProps {
+  onClick: ((newDriver: driversSettings) => void);
+}
+
+function deleteEntry(netid: string, driverList: driversSettings[]) {
+  const newDriverList = [];
+  for (let i = 0; i < driverList.length; i += 1) {
+    if (driverList[i].netid !== netid) {
+      newDriverList.push(driverList[i]);
+    }
+  }
+  return newDriverList;
```

Comment:
Return the result of a call to JavaScript's array filter function instead of coding this explicitly.